src/main/java/frc/robot/subsystems/motorSubsystem/MotorSubsystem.java

```diff
@@ -0,0 +1,72 @@
+package frc.robot.subsystems.motorSubsystem;
+
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.hardware.interfaces.IRequest;
+import frc.robot.hardware.interfaces.InputSignal;
+import frc.robot.subsystems.GBSubsystem;
+
+public class MotorSubsystem extends GBSubsystem {
+
+	private final ControllableMotor motor;
+	private final MotorCommandBuilder commandBuilder;
+	private IRequest<Double> voltageRequest;
+	private InputSignal<Double> voltageSignal;
+	private IRequest lastChangedRequest;
+
+
+	public MotorSubsystem(ControllableMotor motor, String logPath) {
+		super(logPath);
+		this.motor = motor;
+		this.commandBuilder = new MotorCommandBuilder(this);
+	}
+
+	public MotorSubsystem withVoltageControl(IRequest<Double> voltageRequest, InputSignal<Double> voltageSignal) {
+		this.voltageRequest = voltageRequest;
+		this.voltageSignal = voltageSignal;
+		return this;
+	}
+
+	public Double getAppliedVoltage() {
```

Comment:
```suggestion
	public Double getVoltage() {
```
that how all names now